iosxr_setup.py

```diff
@@ -33,49 +36,75 @@ def open_sessions(self, username, password):
         self.xr_telnet_sessions.append(self.xr1)
 
     #
-    # Back to back network config
+    # Configure a vagrant box
     #
     def setup_networking(self, host_ip=None, gateway=None, allow_root_login=False):
         xr1 = self.xr1
 
         #
         # Configure xr
         #
+
+        # ZTP causes some startup issues so disable it during box creation
+        xr1.send("ztp terminate noprompt")
+
+        # Determine if the image is a crypto/k9 image or not
+        # This will be used to determine whether to configure ssh or not
+        xr1.send("run rpm -qa | grep k9sec")
+        time.sleep(2)
+        output = xr1.wait("[\$#]")
+        k9 = re.search(r'iosxrv-k9sec', output)
+
+        # Wait for a management interface to be available
         xr1.repeat_until("sh run | inc MgmtEth",
                          match_txt="interface MgmtEth",
                          debug="Check interface ready",
                          timeout=5)
+
+        # Send conf t until configuration mode is entered
         xr1.wait_xr_conf_mode()
 
+        # Enable telnet
+        xr1.send("telnet vrf default ipv4 server max-servers 10")
+        xr1.wait("config")
+
         # Bring up dhcp on MGMT for vagrant access
         xr1.send("interface MgmtEth0/RP0/CPU0/0")
         xr1.send(" ipv4 address dhcp")
         xr1.send(" no shutdown")
+        xr1.wait("config-if")
 
         # TPA source update
         xr1.send("tpa address-family ipv4 update-source MgmtEth0/RP0/CPU0/0")
+        xr1.wait("config")
 
         if gateway is not None:
             xr1.send("router static address-family ipv4 unicast 0.0.0.0/0 MgmtEth0/RP0/CPU0/0 %s" % gateway)
+            xr1.wait("config")
 
-        xr1.send("ssh server v2")
-        xr1.send("ssh server vrf default")
-        time.sleep(2)  # Give the parser a little time
-        xr1.send("commit")
-        time.sleep(2)  # Give the commit a little time
-        xr1.send("commit")  # Second commit helps when parser is holding onto config buffer
+        # Configure ssh if a k9/crypto image
+        if k9:
+            xr1.send("ssh server v2")
+            xr1.wait("config")
+            xr1.send("ssh server vrf default")
+            xr1.wait("config")
 
+        # Configure GRPC protocol
         xr1.send("grpc")
         xr1.send(" port 57777")
-        xr1.send("telnet vrf default ipv4 server max-servers 10")
-        time.sleep(2)  # Give the parser a little time
+        xr1.wait("config-grpc")
+
+        # Commit changes and end
         xr1.send("commit")
-        time.sleep(2)  # Give the commit a little time
-        xr1.send("commit")  # Second commit helps when parser is holding onto config buffer
+        # A sleep and another commt can help if config locks are seen.
```

Comment:
Will fix the typo. Keeping this code here commented out until I am sure it's stable.